lgc/interface/lgc/Pipeline.h

```diff
@@ -126,71 +126,84 @@ struct Options {
-  unsigned maxThreadGroupsPerComputeUnit = 0;
+  unsigned maxThreadGroupsPerComputeUnit;
 
-  unsigned waveSize = 0;     // Control the number of threads per wavefront (GFX10+)
-  unsigned subgroupSize = 0; // Override for the wave size when the shader uses gl_SubgroupSize, 0 for no override
-  unsigned wgpMode = 0;      // Whether to choose WGP mode or CU mode (GFX10+)
-  WaveBreak waveBreakSize = WaveBreak::None; // Size of region to force the end of a wavefront (GFX10+).
-                                             // Only valid for fragment shaders.
+  unsigned waveSize;       // Control the number of threads per wavefront (GFX10+)
+  unsigned subgroupSize;   // Override for the wave size when the shader uses gl_SubgroupSize, 0 for no override
+  unsigned wgpMode;        // Whether to choose WGP mode or CU mode (GFX10+)
+  WaveBreak waveBreakSize; // Size of region to force the end of a wavefront (GFX10+).
+                           // Only valid for fragment shaders.
 
   // Vector size threshold for load scalarizer. 0 means do not scalarize loads at all.
-  unsigned loadScalarizerThreshold = 0;
+  unsigned loadScalarizerThreshold;
 
   // Use the LLVM backend's SI scheduler instead of the default scheduler.
-  bool useSiScheduler = false;
+  bool useSiScheduler;
 
   // Whether update descriptor root offset in ELF
-  bool updateDescInElf = false;
+  bool updateDescInElf;
 
   // Default unroll threshold for LLVM.
-  unsigned unrollThreshold = 0;
+  unsigned unrollThreshold;
 
   /// Override FP32 denormal handling.
-  DenormalMode fp32DenormalMode = DenormalMode::Auto;
+  DenormalMode fp32DenormalMode;
 
   /// Whether enable adjustment of the fragment shader depth import for the variable shading rate
-  bool adjustDepthImportVrs = false;
+  bool adjustDepthImportVrs;
 
   // Unroll loops by specified amount. 0 is default, 1 is no unroll.
-  unsigned forceLoopUnrollCount = 0;
+  unsigned forceLoopUnrollCount;
 
   // Disable loop unrolling.
-  bool disableLoopUnroll = false;
+  bool disableLoopUnroll;
 
   // Threshold for minimum number of blocks in a loop to disable the LICM pass.
-  unsigned disableLicmThreshold = 0;
+  unsigned disableLicmThreshold;
 
   // Threshold to use for loops with Unroll hint. 0 to use llvm.loop.unroll.full metadata.
-  unsigned unrollHintThreshold = 0;
+  unsigned unrollHintThreshold;
 
   // Threshold to use for loops with DontUnroll hint. 0 to use llvm.loop.unroll.disable metadata.
-  unsigned dontUnrollHintThreshold = 0;
+  unsigned dontUnrollHintThreshold;
+
+  ShaderOptions() {
+    // The memory representation of this struct gets written into LLVM metadata. To prevent uninitialized values from
+    // being written, we force everything to 0, including alignment gaps.
+    memset(this, 0, sizeof(ShaderOptions));
+  }
+
+  ShaderOptions(const ShaderOptions &opts) { *this = opts; }
+
+  ShaderOptions &operator=(const ShaderOptions &opts) {
+    // Copy everything, including data in alignment
```

Comment:
It should not be strictly necessary to copy the alignment gaps here, because they should already be 0 in both source and destination. But it doesn't hurt.